ttnn/cpp/ttnn/operations/transformer/sdpa/device/sdpa_op.cpp

```diff
@@ -21,24 +21,29 @@ void ScaledDotProductAttention::validate(
         TT_FATAL(input_tensor.get_dtype() == DataType::BFLOAT16 || input_tensor.get_dtype() == DataType::BFLOAT8_B);
     }
 
-    auto mask = optional_input_tensors.at(0).value();
-    TT_FATAL(mask.storage_type() == StorageType::DEVICE, "Operands to SDPA need to be on device!");
-    TT_FATAL(input_tensors.at(0).device() == mask.device());
-    TT_FATAL(mask.get_layout() == Layout::TILE);
-    TT_FATAL(mask.get_dtype() == DataType::BFLOAT16 || mask.get_dtype() == DataType::BFLOAT8_B);
+    const auto& mask_option = optional_input_tensors.at(0);
+    if (mask_option.has_value()){
+        auto mask = optional_input_tensors.at(0).value();
+        TT_FATAL(mask.storage_type() == StorageType::DEVICE, "Operands to SDPA need to be on device!");
+        TT_FATAL(input_tensors.at(0).device() == mask.device());
+        TT_FATAL(mask.get_layout() == Layout::TILE);
+        TT_FATAL(mask.get_dtype() == DataType::BFLOAT16 || mask.get_dtype() == DataType::BFLOAT8_B);
```

Comment:
These error messages should say something like, "When mask is provided to SPDA ... [expectation] [exact value provided which is not aligned with expectations]"